### PR TITLE
Actions

### PR DESCRIPTION
**Pull request by nimasmi**
Add GitHub actions

### CODE REVIEW

File: .github/workflows/CI-tests.yml [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    - name: Run image
+      uses: abatilo/actions-poetry@v2.0.0
+      with:
+        poetry-version: ${{ env.poetry-version }}
+    - name: Install Python packages
+      run: |
+          poetry config virtualenvs.in-project true
+          poetry install
+
+  lint:
```
Comment (zerolab) [inline .github/workflows/CI-tests.yml:38]: suggestion: use [pre-commit](https://pre-commit.com/) and the relevant GitHub action

e.g. https://github.com/torchbox/wagtailmedia/blob/main/.github/workflows/test.yml#L14-L22 / https://github.com/torchbox/wagtailmedia/blob/main/.pre-commit-config.yaml

File: .github/workflows/CI-tests.yml [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Cache Poetry virtual env
+      uses: actions/cache@v2
+      env:
+          cache-name: cache-poetry-virtualenv
+      with:
+          path: .venv
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-python-${{ matrix.python-version }}-${{ hashFiles('pyproject.toml') }}
```
Comment (zerolab) [inline .github/workflows/CI-tests.yml:28]: question: do you need `${{ env.cache-name }}` if that's the same everywhere? or is this with a plan to backport to other projects?

**Comment by zerolab** [comment]
@nimasmi you will also be interested in https://packaging.python.org/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/ as a follow-up to this PR

File: .github/workflows/CI-tests.yml (hunk 2 above)
Comment (thibaudcolas) [reply .github/workflows/CI-tests.yml:28]: suggestion: use a hash of the lockfile instead.

```suggestion
          key: ${{ runner.os }}-build-${{ env.cache-name }}-python-${{ matrix.python-version }}-${{ hashFiles('pyproject.lock') }}
```

Same for other cases below.

Comment (nimasmi) [reply .github/workflows/CI-tests.yml]: Thanks. I've fixed these.

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -46,5 +46,5 @@ repos:
     rev: v0.13.0
     hooks:
       - id: detect-secrets
-        args: ['--baseline', '.secrets.baseline']
+        args: ["--baseline", ".secrets.baseline"]
```
Comment (zerolab) [inline .pre-commit-config.yaml:49]: to fix the linting failure change
`repo: git@github.com:Yelp/detect-secrets` to `repo: https://github.com/Yelp/detect-secrets`

**Review by zerolab — APPROVED** [review]
LGTM

**Pull request merged**